fedimint-bin-tests/src/federation.rs

```diff
@@ -78,7 +80,22 @@ impl Federation {
         })
     }
 
-    pub fn members(&self) -> &[Fedimintd] {
+    pub async fn restart_server(
+        &mut self,
+        process_mgr: &ProcessManager,
+        bitcoind: Bitcoind,
+        peer_id: usize,
+    ) -> Result<()> {
+        self.members[peer_id] = Some(Fedimintd::new(process_mgr, bitcoind, peer_id).await?);
+        Ok(())
+    }
```

Comment:
Restarting the server fails, because it is in the process of shutting down from `kill-server()` function. So we need to wait for the child process to completely shut down. I can think of some solutions but they lead to exposing `Child` field inside `ProcessHandleInner` and that defeats the purpose of them being private. What should we do here?

Reply:
add `ProcessHandle::kill()`